experiments/compositions/composition/internal/controller/composition_controller.go

```diff
@@ -74,20 +78,29 @@ func (r *CompositionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		// on deleted requests.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+	oldStatus := composition.Status
+
+	// Try updating status before returning
+	defer func() {
+		if !reflect.DeepEqual(oldStatus, composition.Status) {
+			if err := r.Client.Status().Update(ctx, &composition); err != nil {
+				logger.Error(err, "unable to update Composition status")
+			}
+		}
+	}()
 
 	logger = logger.WithName(composition.Name).WithName(fmt.Sprintf("%d", composition.Generation))
 
 	logger.Info("Validating Compostion object")
 	if !composition.Validate() {
 		logger.Info("Validation Failed")
-		if err := r.Client.Status().Update(ctx, &composition); err != nil {
-			logger.Error(err, "unable to update Composition status")
-			return ctrl.Result{}, err
-		}
+		return ctrl.Result{}, fmt.Errorf("Validation failed")
 	}
 
+	composition.Status.ClearCondition(compositionv1.Error)
```

Comment:
Maybe move this down to where we return no-error in runComposition (so that it's clear who owns / is responsible for this condition)?

Reply:
makes sense. done.